### DISCUSSION TITLE
tons of errors on my RX6700XT

### DISCUSSION THREAD

**Discussion by Kyr4l** [GpuZelenograd/memtest_vulkan#43] [Card-specific memtest_vulkan error logs]
hello,

the card is slightly undervolted (base -85, changed to -120 and it stays stable, went to -110 currently just to be sure)
i've started noticing artifacts in some games (helldivers 2, some random white bars popping on one of my screens for an extremely short time), but no artifacts during benchmarks with unigine heaven
the gpu is power limited to 174W, the max block was lowered from 2644MHz to 2620MHz
i've repasted the card recently and swapped the stock thermal vram pads for brand new pads from artctic, the pads are correctly sized and are as thick as the stock pads

what should i do

[memtest_vulkan.log](https://github.com/user-attachments/files/15812112/memtest_vulkan.log)

**Comment by galkinvv** [comment]
Hello, all errors in your log are "total errors 0x1 out of 0x34000000 (0.00000011%)" and share the same adress range 0x??C3464D0..=0x??C3464D3


It seems that 1 bit in one of the card memory GDDR6 ICs has issues. So, _literally_ you have 99.99999988% of your VRAM ok and  0.00000011% "sometimes produce errors". The mian GPU chip itself looks not to be related. As a workaround you may try lowering memory clock to see is it helps.

Since it is a single-bit error - to visualize the problem in other tests -  you may need to launch several intances of test named "Furmark donut 6500MB GL" from MSI Kombustor test suite (maybe with artifact scanner enabled).

If the GPU is used only for gaming - there is very low error count and if it will not progress over time it can be just ignored unless artifacting is too huge.

However I'd not suggest using it for non-gaming activities like neural networks or some rendering.

**Reply by Kyr4l** [reply]
is it possible to blacklist this faulty bit so the system cant use it ? 
i've ran furmark for a moment and no artifacts were detected
i do use the gpu for AI and video rendering from time to time

**Reply by galkinvv** [reply]
Unfortunately blacklisting a faulty bit is not possible.

There is a ~10-20% chance that reflashing RX6700XT into RX6700 will help - it disables one memory IC, but you can't select which. So you need to be really lucky to have an error in that exact memory that will be disbaled. **However even trying this is risky!** since the RX6700 VBIOS may assume some other models of board power ICs, etc. Don't forget making a backup of your original VBIOS!

**Comment by Kyr4l** [comment]
that's really strange, i've done a system update, ram a memtest86+ and everything went fine
so i ran `memtest_vulkan` again after i realized that after a few tests and chassis fan curve calibration, the overall temps got better. 

i'm guessing the VRAM chips heat allowed the new thermal pads to fit perfectly due to thermal expansion and the material getting softer with heat, which could have allowed it to fit normally. but that's only an hypothesis which i can't truly verify.

the memtest is now passed :

```
https://github.com/GpuZelenograd/memtest_vulkan v0.5.0 by GpuZelenograd
To finish testing use Ctrl+C

1: Bus=0x0C:00 DevId=0x73DF   12GB AMD Radeon RX 6700 XT (RADV NAVI22)
Standard 5-minute test of 1: Bus=0x0C:00 DevId=0x73DF   12GB AMD Radeon RX 6700 XT (RADV NAVI22)
      1 iteration. Passed  0.0623 seconds  written:    6.5GB 290.3GB/sec        checked:    9.8GB 244.5GB/sec
     18 iteration. Passed  1.0600 seconds  written:  110.5GB 287.7GB/sec        checked:  165.8GB 245.2GB/sec
    100 iteration. Passed  5.0408 seconds  written:  533.0GB 293.1GB/sec        checked:  799.5GB 248.1GB/sec
    586 iteration. Passed 30.0361 seconds  written: 3159.0GB 290.7GB/sec        checked: 4738.5GB 247.2GB/sec
   1071 iteration. Passed 30.0250 seconds  written: 3152.5GB 289.5GB/sec        checked: 4728.8GB 247.1GB/sec
   1554 iteration. Passed 30.0421 seconds  written: 3139.5GB 287.9GB/sec        checked: 4709.2GB 246.1GB/sec
   2039 iteration. Passed 30.0332 seconds  written: 3152.5GB 289.4GB/sec        checked: 4728.8GB 247.1GB/sec
   2524 iteration. Passed 30.0242 seconds  written: 3152.5GB 289.5GB/sec        checked: 4728.8GB 247.1GB/sec
   3008 iteration. Passed 30.0446 seconds  written: 3146.0GB 288.6GB/sec        checked: 4719.0GB 246.5GB/sec
   3492 iteration. Passed 30.0218 seconds  written: 3146.0GB 288.7GB/sec        checked: 4719.0GB 246.8GB/sec
   3975 iteration. Passed 30.0021 seconds  written: 3139.5GB 288.8GB/sec        checked: 4709.2GB 246.2GB/sec
   4459 iteration. Passed 30.0401 seconds  written: 3146.0GB 288.9GB/sec        checked: 4719.0GB 246.4GB/sec
Standard 5-minute test PASSed! Just press Ctrl+C unless you plan long test run.
Extended endless test started; testing more than 2 hours is usually unneeded
use Ctrl+C to stop it when you decide it's enough

^C
memtest_vulkan: no any errors, testing PASSed.
  press any key to continue...

```

**Reply by galkinvv** [reply]
While heat-related scenario is possible, here is an argument against it: during one of your tests in original log - some of errors appeared in first 10 seconds, and  even earlier. But cooling system typically does not have any effect suring first 30 seconds.

There is a Nitro 6700XT without fans and without memory heatsink - only with GPU heatsink (it is in my repair lab). I launched test with it and 30 seconds 2 times are passed without any problems (of course it become quite hot, but 30 seconds in such state is fine).

Two times since I tried both RADV and AMDVLK, speeds are similar to yours _last_ log:

RADV:
```
VK_DRIVER_FILES=/usr/share/vulkan/icd.d/radeon_icd.x86_64.json ./memtest_vulkan
Standard 5-minute test of 1: Bus=0x03:00 DevId=0x73DF   12GB AMD Radeon RX 6700 XT (RADV NAVI22)
                     written:    2.8GB  291.9GB/s      checked:    5.5GB  245.2GB/s   
```

AMDVLK:
```
VK_DRIVER_FILES=/usr/share/vulkan/icd.d/amd_icd64.json ./memtest_vulkan
Standard 5-minute test of 1: Bus=0x03:00 DevId=0x73DF   12GB AMD Radeon RX 6700 XT
            written:    7.2GB  296.6GB/s      checked:   10.9GB  256.0GB/s     
```

**But in the _initial_ log** attached to first message - the GB/sec speeds are always a bit slower
```
 written:    6.2GB 269.7GB/sec        checked:    9.4GB 240.8GB/sec
```

So it seems that either something changed during driver update or the slower speed is caused by ECC-corrections and the problem is "unstable"

**Reply by Kyr4l** [reply]
so it's either a driver update or the cooling that has improved, or a mix of both, i guess i'll never know
